src/OpenLoco/Ui/ProgressBar.cpp

```diff
@@ -0,0 +1,123 @@
+#include "ProgressBar.h"
+#include "../Interop/Interop.hpp"
+#include "../Localisation/FormatArguments.hpp"
+#include "../Localisation/StringManager.h"
+#include "../OpenLoco.h"
+#include "WindowManager.h"
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::Ui::ProgressBar
+{
+    static constexpr uint32_t _isInternalWindow = (1 << 31);
+    static uint32_t _1136590;
+    static char _captionBuffer[256] = {};
+
+    // 0x004CF5DA
+    static void loadingWindowCreate(string_id captionStringId)
+    {
+        // TODO: platform-specific implementation?
+        registers regs;
```

Comment:
I feel we could at least do the format string part and call at 4cf5ee

Reply:
I'd looked at doing the format string bit before, but hadn't done it because I can't test the Windows progress bars. Moreover, it's pushing the address for `buffer_string_format` on the stack, which I'd already replaced with a local buffer, `captionString`.